tests/unit/test_action_serialization.py

```diff
@@ -21,6 +21,8 @@
 
 
 def serialization_deserialization(original_action_dict, cls):
+    if 'source' not in original_action_dict:
+        original_action_dict['source'] = 'user'
     action_instance = event_from_dict(original_action_dict)
```

Comment:
Same here, 'agent' if anything

Reply:
I'm wondering why this is necessary? Shouldn't the source being set from elsewhere (e.g., Controller or where the action is created?)